protocol.md

```diff
@@ -1,55 +1,248 @@
+| N bytes | Packed derivation path used for the FROM coin                                                                           |
+
+#### START_SIGNING_TRANSACTION
+
+No data expected.
+
+
+## Response:
+
+| Bytes         | Description           |
+| ------------- | --------------------- |
 | N bytes       | Command specific data |
-| 2 bytes       | Result |
-
-Result can be one of the following sequences:
-| Bytes  | Name                     | Description                                      |
-| ------ | ------------------------ | ------------------------------------------------ |
-| 0x6A80 | INCORRECT_COMMAND_DATA   | Command byte, or specific command data is incorrect |
-| 0x6A81 | DESERIALIZATION_FAILED   | Can't parse partner transaction |
-| 0x6A82 | WRONG_TRANSACTION_ID     | Transaction ID is not equal to one generated on the first step|
-| 0x6A83 | INVALID_ADDRESS          | Refund or payout address doesn't belong to us |
-| 0x6A84 | USER_REFUSED             | User refused transaction |
-| 0x6A85 | INTERNAL_ERROR           | Some error inside the device application |
-| 0x6E00 | CLASS_NOT_SUPPORTED      | |
-| 0x6D00 | INVALID_INSTRUCTION      | Command byte is not the one from the "Possible commands" table |
-| 0x9D1A | SIGN_VERIFICATION_FAIL   | Signature is incorrect |
-| 0x9000 | SUCCESS                  | Success |
-
-## Start new transaction
-This should be the first command
-On this step device will generate 10 capital latin letters and return it to the the host application.
-
-No command data expected in this command request.
-
-In case of success command specific data in response contains 10 bytes - representation of transaction ID.
-
-## Set partner key
-Partner public key and partner name signed by Ledger private key is send in this command. This command is only allowed after successful return from `START_NEW_TRANSACTION_COMMAND`
+| 2 bytes       | Return code           |
+
+### Returned data
+
+Only the COMMANDS described below return data. All others only return a code.
+
+#### GET_VERSION
+
+In case of success, this command will return the application version in format {MAJOR.MINOR.PATCH}.
+
+#### START_NEW_TRANSACTION
+
+In case of success, this command will return a nonce called "device transaction id" used for the initiated flow. \
+For TYPE SWAP_LEGACY, the format of this nonce is a 10 char array. \
+For all other TYPES, the format of this nonce is a 32 bytes array.
+
+### Return code
+
+Return code can be one of the following values:
+
+| Bytes  | Name                     | Description                                                                              |
+| ------ | ------------------------ | ---------------------------------------------------------------------------------------- |
+| 0x6A80 | INCORRECT_COMMAND_DATA   | Command byte, or specific command data is incorrect                                      |
```

Comment:
"Command byte incorrect" is not very specific, what does it mean?
Given the context, an "incorrect command" can also be responded with "UNEXPECTED_INSTRUCTION" or "INVALID_INSTRUCTION". I don't know what else it could be here.

Reply:
Replaced by
`The DATA sent does not match the correct format for the COMMAND specified`